File: brandlyandco/src/main/java/com/ecommerce/brandlyandco/entity/Product.java
package com.ecommerce.brandlyandco.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String productname;
    private double  specialprice;
    private double regularprice;
    private int quantity;
    private String model;
    private int rating;
    private String description;
    private  int warranty;
    private String salesservice;
    private String policy;
    private double offer;
    private  String imagea;
    private String imageb;
    private String imagec;
    private String imaged;


    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "catagory_id")
    private Category catagory;



}

File: flutter solve/capnest/lib/homepage/model/Product.dart
import 'Category.dart';

class Product {
  final int id;
  final String productname;
  final double specialprice;
  final double regularprice;
  final int quantity;
  final String model;
  final int rating;
  final String description;
  final int warranty;
  final String salesservice;
  final String policy;
  final double offer;
  final String imagea;
  final String imageb;
  final String imagec;
  final String imaged;
  final Category? category;

  Product({
    required this.id,
    required this.productname,
    required this.specialprice,
    required this.regularprice,
    required this.quantity,
    required this.model,
    required this.rating,
    required this.description,
    required this.warranty,
    required this.salesservice,
    required this.policy,
    required this.offer,
    required this.imagea,
    required this.imageb,
    required this.imagec,
    required this.imaged,
    this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] ?? 0,
      productname: json['productname'] ?? '',
      specialprice: (json['specialprice'] ?? 0).toDouble(),
      regularprice: (json['regularprice'] ?? 0).toDouble(),
      quantity: json['quantity'] ?? 0,
      model: json['model'] ?? '',
      rating: json['rating'] ?? 0,
      description: json['description'] ?? '',
      warranty: json['warranty'] ?? 0,
      salesservice: json['salesservice'] ?? '',
      policy: json['policy'] ?? '',
      offer: (json['offer'] ?? 0).toDouble(),
      imagea: json['imagea'] ?? '',
      imageb: json['imageb'] ?? '',
      imagec: json['imagec'] ?? '',
      imaged: json['imaged'] ?? '',
      category: json['catagory'] != null
          ? Category.fromJson(json['catagory'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'productname': productname,
      'specialprice': specialprice,
      'regularprice': regularprice,
      'quantity': quantity,
      'model': model,
      'rating': rating,
      'description': description,
      'warranty': warranty,
      'salesservice': salesservice,
      'policy': policy,
      'offer': offer,
      'imagea': imagea,
      'imageb': imageb,
      'imagec': imagec,
      'imaged': imaged,
      'catagory': category?.toJson(),
    };
  }
}